components/scream/cmake/tpls/Scorpio.cmake

```diff
@@ -56,10 +56,8 @@ macro (CreateScorpioTarget CREATE_FLIB)
       # Not a CIME build. Add scorpio as a subdir
       add_subdirectory (${E3SM_EXTERNALS_DIR}/scorpio ${CMAKE_BINARY_DIR}/externals/scorpio)
       EkatDisableAllWarning(pioc)
+      EkatDisableAllWarning(piof)
```

Comment:
I think mistakes were made here, `pioc` warnings were disabled twice  if CREATE_FLIB is on. I think maybe the original coder intended to disable `piof` if CREATE_FLIB is on, but that's not right either because `piof` gets built even if CREATE_FLIB is off.

Reply:
`piof` should not be created unless CREATE_FLIB is on.

Reply:
Ah, wait, it appears scorpio _does_ build flib by default. I believe then the correct code should have been:

```
add_subdirectory(...)
EkatDisableAllWarning(pioc)
if (PIO_ENABLE_FORTRAN) # ON by default in scorpio
  EkatDisableAllWarning(piof)
endif()
```
Without the if, someone _might_ disable f90 scropio (if we ever support pure C scorpio), and then cmake could complain that piof is not a target. Alternatively, we might change the if to the following
```
if (TARGET piof)
...
endif()
```